lib/superstore/attribute_methods/definition.rb

```diff
@@ -7,10 +7,16 @@ def initialize(model, name, type_class, options)
         @type = type_class.new(model, options)
       end
 
-      def instantiate(value)
-        return if value.nil?
+      def encode(value)
+        type.encode(value) unless value.nil?
+      end
+
+      def decode(value)
+        type.decode(value) unless value.nil?
+      end
 
-        value.kind_of?(String) ? type.decode(value) : type.typecast(value)
+      def typecast(value)
+        type.typecast(value) unless value.nil?
       end
```

Comment:
`instantiate` served a dual purpose and was confusing. It supported both reading from the db and converting user input.